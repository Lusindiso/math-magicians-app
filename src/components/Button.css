.button {
  background-color: #e0e0e0;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 1rem;
  border: 2px solid #cecece;
}

.btn {
  grid-column: 1/3;
}
