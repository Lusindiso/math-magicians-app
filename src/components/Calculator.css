.calculator {
  margin: 5rem auto;
  width: 50%;
  height: 62vh;
  border: 1px solid white;
  border-top-width: 4px;
  display: grid;
}

.bottom {
  height: 45rem;
  display: grid;
  grid-template-areas:
    'button button button last'
    'button button button last'
    'button button button last'
    'button button button last'
    'button button button last';
}
