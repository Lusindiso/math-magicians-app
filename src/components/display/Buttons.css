.buttons {
  grid-area: button;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
}
