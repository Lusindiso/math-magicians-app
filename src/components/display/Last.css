.last {
	grid-area: last;
	display: grid;
	grid-template-columns: 1fr;
	grid-template-rows: repeat(5, 1fr);
}

.last > * {
	background-color: #f5913e;
	display: flex;
	justify-content: center;
	align-items: center;
	padding: 1rem;
}
