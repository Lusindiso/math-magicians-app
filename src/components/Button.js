/* eslint-disable react/prop-types */
import React, { PureComponent } from 'react';
import './Button.css';

class Button extends PureComponent {
	render() {
		const { value } = this.props;
		return <div className='button'>{value}</div>;
	}
}

export default Button;
