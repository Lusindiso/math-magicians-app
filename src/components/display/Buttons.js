import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';

import './Buttons.css';

const list = [
  'AC',
  '+/-',
  '%',
  '7',
  '8',
  '9',
  '4',
  '5',
  '6',
  '1',
  '2',
  '3',
  '0',
  '.',
];

class Buttons extends PureComponent {
  render() {
    const { handleClick } = this.props;

    return (
      <div className="buttons">
        {list.map((el) => (
          <Button value={el} key={el} handleClick={handleClick} />
        ))}
      </div>
    );
  }
}

Buttons.propTypes = {
  // eslint-disable-next-line react/require-default-props
  handleClick: PropTypes.func,
};

export default Buttons;
