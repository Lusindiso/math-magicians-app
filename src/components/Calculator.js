import React, { Component } from 'react';
import Buttons from './display/Buttons';
import './Calculator.css';
import Last from './display/Last';
import Screen from './Screen';
import calculate from '../logic/calculate';

class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      total: 0,
      next: 0,
      operation: null,
    };
    this.handleClick = this.handleClick.bind(this);
  }

	handleClick = (btnName) => {
	  this.setState((prevState) => calculate(prevState, btnName));
	};

	render() {
	  const { total, next, operation } = this.state;
	  return (
  <div className="calculator">
    <Screen total={total} next={next} operation={operation} />
    <div className="bottom">
      <Last handleClick={this.handleClick} />
      <Buttons handleClick={this.handleClick} />
    </div>
  </div>
	  );
	}
}

export default Calculator;
