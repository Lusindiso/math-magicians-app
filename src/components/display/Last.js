import React, { PureComponent } from 'react';
import Button from '../Button';
import './Last.css';

const list = ['รท', 'X', '-', '+', '='];
class Last extends PureComponent {
	render() {
		return (
			<div className='last'>
				{list.map((el) => (
					<Button value={el} key={el} />
				))}
			</div>
		);
	}
}

export default Last;
