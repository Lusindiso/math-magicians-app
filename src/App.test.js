import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';
import calculate from './logic/calculate';
import operate from './logic/operate';

import Calculator from './components/Calculator';
import Home from './components/pages/Home';
import Quotes from './components/pages/Quote';
import CalculatorPage from './components/pages/CalculatorPage';
import Buttons from './components/display/Buttons';
import Last from './components/display/Last';
import NavBar from './components/UI/Navbar';
import Header from './components/UI/Header';
import Screen from './components/Screen';
import Button from './components/Button';

describe('Functions test', () => {
  test('calculate.js test', () => {
    const initialItems = {
      total: 0,
      next: null,
      operation: null,
    };
    const buttonName = '=';
    if (buttonName === '.') {
      expect(calculate(initialItems, buttonName)).toEqual(
        { total: 0, next: null, operation: null },
      );
    } else if (buttonName === '.') {
      expect(calculate(initialItems, buttonName)).toEqual(
        { total: 0.0, next: null, operation: null },
      );
    } else if (buttonName === '1') {
      expect(calculate(initialItems, buttonName)).toEqual(
        { total: null, next: '1' },
      );
    } else if (buttonName === '=') {
      expect(calculate(initialItems, buttonName)).toEqual({});
    }
  });

  test('operate.js addition test', () => {
    const numberOne = '1';
    const numberTwo = '2';
    const operation = '+';

    expect(operate(numberOne, numberTwo, operation)).toBe('3');
  });

  test('operate.js subtraction test', () => {
    const numberOne = '1';
    const numberTwo = '2';
    const operation = '-';

    expect(operate(numberOne, numberTwo, operation)).toBe('-1');
  });

  test('operate.js multiplication test', () => {
    const numberOne = '1';
    const numberTwo = '2';
    const operation = 'x';

    expect(operate(numberOne, numberTwo, operation)).toBe('2');
  });

  test('operate.js division test', () => {
    const numberOne = '1';
    const numberTwo = '2';
    const operation = 'รท';

    expect(operate(numberOne, numberTwo, operation)).toBe('0.5');
  });

  test('operate.js division by 0 test', () => {
    const numberOne = '1';
    const numberTwo = '0';
    const operation = 'รท';

    expect(operate(numberOne, numberTwo, operation)).toBe('Can\'t divide by 0.');
  });

  test('operate.js remainder test', () => {
    const numberOne = '5';
    const numberTwo = '2';
    const operation = '%';

    expect(operate(numberOne, numberTwo, operation)).toBe('1');
  });

  test('operate.js remainder when divide by 0 test', () => {
    const numberOne = '1';
    const numberTwo = '0';
    const operation = '%';

    expect(operate(numberOne, numberTwo, operation)).toBe('Can\'t find modulo as can\'t divide by 0.');
  });
});
