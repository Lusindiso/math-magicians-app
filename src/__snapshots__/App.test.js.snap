// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component test Button component should perform correctly on initial run 1`] = `
<button
  className="button "
  onClick={[Function]}
  type="button"
/>
`;

exports[`component test Buttons component should perform correctly on initial run 1`] = `
<div
  className="buttons"
>
  <button
    className="button "
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="button "
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="button "
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="button "
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="button "
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="button "
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="button "
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="button "
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="button "
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="button "
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="button "
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="button "
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="button btn"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    className="button "
    onClick={[Function]}
    type="button"
  >
    .
  </button>
</div>
`;

exports[`component test Calculator component should perform the calculations correctly on initial run 1`] = `
<div
  className="calculator"
>
  <div
    className="screen"
  >
    <p>
      0
    </p>
    <p
      className="operation"
    />
  </div>
  <div
    className="bottom"
  >
    <div
      className="last"
    >
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
    <div
      className="buttons"
    >
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="button btn"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        .
      </button>
    </div>
  </div>
</div>
`;

exports[`component test CalculatorPage component should perform correctly on initial run 1`] = `
<main
  style={
    Object {
      "display": "flex",
      "margin": "8rem 10%",
    }
  }
>
  <p>
    Let's do some maths
  </p>
  <div
    className="calculator"
  >
    <div
      className="screen"
    >
      <p>
        0
      </p>
      <p
        className="operation"
      />
    </div>
    <div
      className="bottom"
    >
      <div
        className="last"
      >
        <button
          className="button "
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
        <button
          className="button "
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          className="button "
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          className="button "
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="button "
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
      <div
        className="buttons"
      >
        <button
          className="button "
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="button "
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="button "
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="button "
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="button "
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="button "
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="button "
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="button "
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="button "
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="button "
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="button "
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="button "
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="button btn"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="button "
          onClick={[Function]}
          type="button"
        >
          .
        </button>
      </div>
    </div>
  </div>
</main>
`;

exports[`component test Header component should perform correctly on initial run 1`] = `
<header
  className="header"
>
  <h1>
    Math Magicians
  </h1>
  <nav>
    <ul
      className="nav"
    >
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className=""
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          className=""
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`component test Home component should perform correctly on initial run 1`] = `
<main>
  <h2
    style={
      Object {
        "margin": "50px 10%",
      }
    }
  >
    Welcome to our page!
  </h2>
  <p
    style={
      Object {
        "margin": "25px 10%",
      }
    }
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Earum pariatur molestias officia, ducimus vitae voluptas asperiores porro provident, iure et deserunt alias iste voluptatum dolorem maxime temporibus inventore. Quidem, odit.
  </p>
  <p
    style={
      Object {
        "margin": "25px 10%",
      }
    }
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Temporibus accusantium enim aliquid officia vero a cumque repellendus doloremque mollitia exercitationem repudiandae suscipit, vitae, voluptatum eveniet tempora architecto alias quis aperiam?
  </p>
</main>
`;

exports[`component test Last component should perform correctly on initial run 1`] = `
<div
  className="last"
>
  <button
    className="button "
    onClick={[Function]}
    type="button"
  >
    รท
  </button>
  <button
    className="button "
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    className="button "
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="button "
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="button "
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;

exports[`component test NavBar component should perform correctly on initial run 1`] = `
<nav>
  <ul
    className="nav"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className=""
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;

exports[`component test Quote component should perform correctly on initial run 1`] = `
<main
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "90vh",
      "justifyContent": "center",
      "margin": "0 10%",
      "width": "90vw",
    }
  }
>
  <p>
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. - William Paul Thurston
  </p>
</main>
`;

exports[`component test Screen component should perform correctly on initial run 1`] = `
<div
  className="screen"
>
  <p />
  <p
    className="operation"
  />
</div>
`;
